### PR TITLE
Add Traffic Ops Golang deliveryservicesregexes

### PR DESCRIPTION
**Pull request by rob05c**

### CODE REVIEW

**Comment by rob05c** [comment]
This is ready, in terms of functionality. I'm not sure how to write API tests for it though. Everything is ID based, deliveryservice_regex doesn't have a non-ID primary key.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1523/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1524/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1525/
Test PASSed.

File: traffic_ops/traffic_ops_golang/routes.go [hunk 1]
```diff
@@ -214,6 +215,14 @@ func Routes(d ServerData) ([]Route, []RawRoute, http.Handler, error) {
 		{1.3, http.MethodPost, `servers/{id}/deliveryservices$`, server.AssignDeliveryServicesToServerHandler(d.DB), auth.PrivLevelOperations, Authenticated, nil},
 		{1.3, http.MethodGet, `servers/{host_name}/update_status$`, server.GetServerUpdateStatusHandler(d.DB), auth.PrivLevelReadOnly, Authenticated, nil},
 
+		//Delivery Services Regexes
+		{1.1, http.MethodGet, `deliveryservices_regexes/?(\.json)?$`, deliveryservicesregexes.Get(d.DB), auth.PrivLevelReadOnly, Authenticated, nil},
+		{1.1, http.MethodGet, `deliveryservices/{dsid}/regexes/?(\.json)?$`, deliveryservicesregexes.DSGet(d.DB), auth.PrivLevelReadOnly, Authenticated, nil},
+		{1.1, http.MethodGet, `deliveryservices/{dsid}/regexes/{regexid}?(\.json)?$`, deliveryservicesregexes.DSGetID(d.DB), auth.PrivLevelReadOnly, Authenticated, nil},
+		{1.1, http.MethodPost, `deliveryservices/{dsid}/regexes/?(\.json)?$`, deliveryservicesregexes.Post(d.DB), auth.PrivLevelReadOnly, Authenticated, nil},
```
[the remark this answers is not in the document]
Comment (rob05c) [reply traffic_ops/traffic_ops_golang/routes.go]: Fixed

**Comment by mitchell852** [comment]
I tested all these Go routes through the TP and they all work fine.

@DylanVolz @dangogh @dewrich - want to give it a quick once over from a Go perspective? It looked fine to me but I'm hardly a Go expert.

**Comment by dewrich** [comment]
```
PASS | ReadHandler (0.00s)
START| UpdateHandler
FAIL | --- FAIL: TestUpdateHandler (0.00s)
     | 	shared_handlers_test.go:251: Expected body {"response":{"ID":1},"alerts":[{"text":"tester was updated.","level":"success"}]} got {"response":[{"ID":1}],"alerts":[{"text":"tester was updated.","level":"success"}]}
```

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1550/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1553/
Test PASSed.

**Comment by mitchell852** [comment]
@rob05c - can you resolve conflicts and i'll merge

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1593/
Test PASSed.

**Pull request merged**